### ISSUE TITLE
What do we want from a Virtual Tabletop?

### ISSUE THREAD

**Issue by crimsonknave** [our-stories/west-marches#43]
There are quite a few options out there for virtual tabletops. https://www.dnd-compendium.com/dm-resources/virtual-tabletops has a quick comparison of many options. Most of these are trying to do more than we need since they are often trying to be the only tool you need. Let me know if you think I missed anything!

### TL;DR

As things currently stand, I think we could probably use the free version of Astral or Tableplop and it would do what we need it to do.

### What we want/need
* Maps with a grid/hex
* Tokens
  * Ideally the ability for players to move their own tokens
* Free features that cover what we need or support for multiple GMs (without each having to pay)
* Initiative tracking?

### What we don't need
* Rolling
* Character Sheets
* 3D: I would prefer a 2D setup as it'll be simpler

### Options
* Roll20
  * Has the features we need
  * Can be slow/laggy 
  * Sometimes gets in the way as it's trying to do more than we need
  * Does someone need to pay?
* Astral Tabletop
  * Has the features we need
  * Free tier seems like it would be fine, subscriptions gives more tokens, effects and storage space
  * Also will be doing more than we need
* Tableplop
  * Based on video, has what we need
  * Free (funded by patreon)
* Let's Role
  * In Alpha, couldn't find a video of the full features
* Foundry VTT
  * GM must host the server, everyone else connects via browser
  * GM must pay
* Fantasy Grounds
  * GM must purchase
* Tabletop Simulator
  * Everyone must purchase
* TaleSpire
  * 3D map maker that others can connect to

**Comment by djigneo** [comment]
I can see some advantages to using Roll20.
Myself and quite probably other GMs have experience in it. I already know a bunch of tricks and tips, so it's not a new tool that the one running the game needs to learn.

I also have quite a few little things that are really nice quality of life things in Roll 20.

- Monster's manual for drag and drop enemies without having to enter stat blocks.
- Automatic HP calculation and tracking on tokens - just drag and drop.
- Script to mark enemies as bloody /dead based on HP.
- Script to automatically number groups of tokens.

I would imagine players would just use the "hide sidebar" option (click the little three lines in the upper left of the tabletop) to only see the tabletop, and the GM would assign the tokens / sheets and let the players move tokens, use ruler, etc.

**Comment by zmoazeni** [comment]
I feel like you both have some strong feelings that a Virtual Table Top is required. And I'm trying to understand that with my experiences of Games that I have run in the past (most have been a combination of discord+Roll20).

**I (as a GM or a Player)** do not need a VTT at all to enjoy this game online (with or without battlemaps). Even more so, if I didn't have to DO ANYTHING as a Player other than communicate where I want to move and what I'm attacking, that feels like more of a win in my book than having to pick up yet another tool.

In fact, I am thinking about starting in the opposite direction to begin with the lowest tech and move increasingly towards the VTT from there. _If it's needed._

A few options that are not on your list:

* [Google Draw](https://docs.google.com/drawings/). Super low tech, but [it works](https://docs.google.com/drawings/d/1Ba0j-Vfr8Z_p__CPtOqIhv8gNUYi--J4wBIIF_QCj1g/edit?usp=sharing).
* [Krita](https://krita.org/en/) even better than Google Draw. Or any other vector graphics program that is screenshared. 
* [MapTool](https://www.rptools.net/toolbox/maptool/). More advanced and starting to get closer to the VTT arena. Can have folks connect and manage their own tokens.

If everyone can see, it really does feel like it will satisfy the base needs. A/V tools like Zoom have annotation abilities so it's pretty easy to deal with the "shared pointer" idea.

Roll20's performance really does bug me. Its lagginess is unavoidable (from what I can tell). I find their tools take some time to grok (the different layers) and even the way you can import and manage tokens isn't intuitive.

If we are relying on outside tools to manage the character sheets, and trust GMs/Players to not lie about dice, why do we actually need a VTT?

I would like to experiment here before we try to force ourselves into a VTT. I honestly think that VTTs shine more with campaign-style games than a loose and open community like this one. I suspect we're going to struggle trying to fit everyone who comes here into a VTT box.

That's my perspective, but I can only control the games I run. I by no means will discourage you all from running games with a VTT. But I do want to hold off on making a community decision here.

**Comment by djigneo** [comment]
Yes. I agree that there's not a driving need for us to make a community decision. Especially with the integration between dnd beyond and discord / Roll20, I think that solved the biggest gaps we had with information being in too many places. (Thanks to @crimsonknave for the legwork on that one.)

I think it's fine to let each GM choose their own tabletop technology (if they even deem a tabletop necessary).
I do agree with the performance concerns on Roll20.

**Comment by crimsonknave** [comment]
I do think there is value in having a set of standard tools. It's going to slow things down if every GM has a different setup and players need to learn/context switch for every game. I don't have strong preferences on what's required. I mostly wanted to start the conversation on what the minimum is and work from there. 

It sounds like the minimum is "A bit more than theater of the mind" So, we need a place to have a map and a way for folks to move around on there. Which is mostly what I laid out in the issue. Given that there are two solutions that appear to have what we want for free is there a reason we don't want to try them and see if they're going to work?

I guess my concern for a GM controlled and shared screen is that there are a lot of places where things can slow down because multiple players need to do things related to location and range. It just seems like it's putting more on the GM when they are already a bottleneck of focus. Having a simple map where everyone can move their tokens solves that for me.

**Comment by zmoazeni** [comment]
@crimsonknave credit where credit is due. Thank you for both starting this discussion AND finding and setting up Beyond20/Discord. (Not to mention you were the first person who pushed me to dndbeyond in the first place). Don't take my thoughts as discouraging this discussion from continuing.

So seriously. I really do appreciate you driving the tooling. :heart:

I personally just need to sit with this for a bit. I need to tinker and experiment with some games in a few systems. 

My driving goal here is the least amount of work for the players. And that is including signing up, logging in, learning, and using any system used.

2nd, and very close, is ease of use (and cost) for GMs too. I want to setup a set of recommendations with the barrier of entry so low the technology isn't the blocker for someone who is interested in dabbling their toes into GMing. Because more likely than not, it'll be those same Players volunteering. 

GMing is tough enough as it is without having to pay and grok a tool just to do it you know? And we all agree all tools listed here have drawbacks.

I think I'm just optimizing for different concerns here.

And. My mind is still very open here. You definitely swayed me in favor of dndbeyond, so there a _solid_ chance I'll be persuaded with map tooling too. ❤
